Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -74,46 +74,48 @@ theorem _root_.ULift.down_algebraMap (r : R) : (algebraMap R (ULift A) r).down =
 
 end ULift
 
-section SubsemiringAlgebra
+section SubsemiringClass
+variable [SetLike S R] [SubsemiringClass S R]
 
-variable {C : Type*} [SetLike C R] [SubsemiringClass C R]
+instance ofClass (s : S) : Algebra s A := .compHom _ (SubsemiringClass.subtype s)
 
-/-- Algebra over a subsemiring. This builds upon `Subsemiring.module`. -/
```

Comment:
Will you generalize [Subsemiring.module](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Ring/Subsemiring/Basic.html#Subsemiring.module) too?

Reply:
I could, but before I do so could you say whether you believe this refactor is a good idea?